packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from pathlib import Path
+from typing import Any, Callable, Coroutine, Type
+
+from pydantic import BaseModel, Field
+
+from ..node_ports.dbmanager import DBManager
+from ..node_ports.exceptions import PortNotFound, UnboundPortError
+from .links import ItemConcreteValue
+from .port_utils import is_file_type
+from .ports_mapping import InputsList, OutputsList
+
+log = logging.getLogger(__name__)
+
+
+class Nodeports(BaseModel):
+    internal_inputs: InputsList = Field(..., alias="inputs")
+    internal_outputs: OutputsList = Field(..., alias="outputs")
+    db_manager: DBManager
+    node_uuid: str
+    save_to_db_cb: Callable[["Nodeports"], Coroutine[Any, Any, None]]
+    node_port_creator_cb: Callable[[DBManager, str], Coroutine[Any, Any, "Nodeports"]]
+    auto_update: bool = False
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def __init__(self, **data: Any):
+        super().__init__(**data)
+        # let's pass ourselves down
+        for input_key in self.internal_inputs:
+            self.internal_inputs[input_key]._node_ports = self
+        for output_key in self.internal_outputs:
+            self.internal_outputs[output_key]._node_ports = self
+
+    @property
+    async def inputs(self) -> InputsList:
+        log.debug("Getting inputs with autoupdate: %s", self.auto_update)
+        if self.auto_update:
+            await self._auto_update_from_db()
+        return self.internal_inputs
+
+    @property
+    async def outputs(self) -> OutputsList:
+        log.debug("Getting outputs with autoupdate: %s", self.auto_update)
+        if self.auto_update:
+            await self._auto_update_from_db()
+        return self.internal_outputs
+
+    async def get(self, item_key: str) -> ItemConcreteValue:
+        try:
+            return await (await self.inputs)[item_key].get()
+        except UnboundPortError:
+            # not available try outputs
+            pass
+        # if this fails it will raise an exception
+        return await (await self.outputs)[item_key].get()
+
+    async def set(self, item_key: str, item_value: ItemConcreteValue) -> None:
```

Comment:
so here the idea is try to set first inputs and if you cannot, try outputs ... 
TIP: the code can be simplified as
```python
async def set(self, ..)-> None:
   try:
       await (await self.inputs)[item_key].set(item_value)
   except UnboundPortError: #no inputs, try outputs
       await (await self.outputs)[item_key].set(Item_value)
```